core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -55,4 +56,28 @@ private SnapshotUtil() {
     }
     return ancestorIds;
   }
+
+  public static boolean isCurrentAncestor(TableMetadata tableMeta, long snapshotId) {
```

Comment:
The other methods in `SnapshotUtil` use the public API, which does not include `TableMetadata`. Can we update this to use `Table` instead of `TableMetadata` as well?